### ISSUE TITLE
Add i18n to VS Code in browser

### ISSUE THREAD

**Issue by brendangadd** [StatCan/kubeflow-containers-desktop#47]
* Have the French language pack installed by default
* Load French UI by default when the browser requests it

**Comment by Jose-Matsuda** [comment]
In dev branch; 
French Language pack UI for VsCode (VsCode in-browser as well as VsCode in the remote desktop) is now installed by default and is found under "Extensions". To switch display language VsCode must restart. 

Running **locally**, I have been able to get the remote vscode to **use** the French UI, for some reason the in-browser vscode editor doesn't like it when you restart it (just keeps using the Eng UI). *Edit: May be due to `code-server is being accessed over an insecure web domain. Web views, the clipboard, and other functionality will not work as expected`  
Running off of one of the notebook servers and installing the language pack manually I was able to get both remote and in-browser to load and use the French UI. 

Need to dive deeper to find where and how to make changes to use browser language (supposing `Accept-Language` like in the Jupyter Web App portion is the key)

**Comment by Jose-Matsuda** [comment]
Can read if you like, nothing too big, moreso so I know I wasn't dreaming. 
**This is all done LOCALLY ofc**
Observing some weird behaviour that I wasn't expecting. 
Made changes to the command in `vscode.conf` to include `--locale=fr`

The first one did not have the `vscode.conf` changed and when I tried changing the language it did not work on restart. The second `docker run` had the `vscode.conf` changed and upon restart the UI did change
Video of it in action [here](https://streamable.com/60ztw9)


Found and looked into the cli for codeserver [here](https://github.com/cdr/code-server/blob/9bde62fbd611a7a91c5f327fa43e0d06f1379169/src/node/cli.ts) and decided perhaps `--locale="fr"` should be used (just checked does not seem to matter).
Ok well here's another video because it's a tad [funky](https://streamable.com/teg4cy)

TLDR; For whatever reason if I do not do `--locale="fr"` in the vscode.conf, the language does not want to change at all even after the restart. If I provide `--locale="fr"` then after updating to 1.50.2 (I was unsure of what version was supported) 

TODO tomorrow:
Update the lang pack to be 1.50.2 (so there's no update shenanigans at the beginning like in the second video) and check that a restart / open and close of the vscode in browser is no longer necessary. 
IFF that works, then I guess will pass a variable that has the browser language to the `--locale=?` 

Update from 'tomorrow'
- Upgraded to 1.50.2, but it does not auto go to FR or appear in the `Select Display Language` as seen [here](https://imgur.com/a/YjxDkM0)
- It recognizes the French pack as installed and enabled when searching for extensions
- Upon installing another language pack and then "restarting" to use that language pack it THEN goes into French and cannot be changed from there.

**Comment by Jose-Matsuda** [comment]
No need to read. summary:
When I by default (install) use the language pack of 1.50.2 (because when I used 1.46.5 it would auto-update on start)
it is NOT recognized as a Display Language (appears in extensions but does not appear as an option to switch). 
-With this instance, you can uninstall then install the extension and it will be recognized. 

Using the 'Config Display Language' Option at least locally seems to do nothing at all, it's suppossed
to modify the `argv.json` file but it does not, tested with a notebook server it works and modifies.

Making use of the option `--locale="fr"` in the vscode.conf file hard sets the language to French.
Modifying the json file at that stage will do nothing (as expected I suppose)


Using either the `vscode.conf` or `argv.json` way to set the language
I'm in a bit of a pickle. If I use 1.46.5 it updates to 1.50.2 and requires a restart. If I use 1.50.2 it is NOT recognized and need to re-install it anyways (and then to use it ofc a restart). 

This seems slightly trivial in the face of actually getting the browser language and then using it to change the VSCode display language since `vscode.conf` I think is configured at the time the container is created / booted up to run (so this is not liked). So perhaps a solution is to make a script that runs as you click VSCode that creates (since we do not create one by default, maybe we should with just "locale"="en") the `argv.json` at home/jovyan/.config/Code/User

**Comment by Jose-Matsuda** [comment]
Just want opinions. 

As the comment says above, vscode preferences are set through the `argv.json` at home/jovyan/.config/Code/User. 
So in order to create that file I would use nodejs because regular javascript is no bueno for file creation. 

I would listen on port 8888 (as thats the landing page I believe), get the accept-language and then make the change to the file to either `en` or `fr` (just need to figure out how to get it going as well)

@wg102 @ca-scribner

**Comment by wg102** [comment]
IMHO, this is the next step to do.
As we previously discussed the `argv.json` file seems to be where you need to do the change. 

Do make sure it finds the accept-language, with the possibility of having none, or that the one selected is neither fr nor eng. 

The only thing would be to decide if we only check when the user hits the dashboard (keeping in mind this issue is directly linked with https://github.com/StatCan/kubeflow-containers-desktop/issues/46 where that dashboard page will need the browser language as well), or for all possible urls that will be hit by kubeflow-container-desktop/

**Comment by Jose-Matsuda** [comment]
Ok looks like I need _1_ more file demonstrated by this [video](https://streamable.com/vyba26)
Of note is also how the extension does NOT try to auto update (which is nice and what I wanted) but I'm pretty sure I didn't do anything for that

Ok seems like we're in the clear. [Video](https://streamable.com/2pvpm8)
